druid/src/command.rs

```diff
@@ -59,31 +50,27 @@ pub struct Selector(&'static str);
 ///
 /// [`Command::new`]: #method.new
 /// [`Command::get_object`]: #method.get_object
+/// [`SingleUse`]: struct.SingleUse.html
 /// [`Selector`]: struct.Selector.html
 #[derive(Debug, Clone)]
 pub struct Command {
     /// The command's `Selector`.
     pub selector: Selector,
-    object: Option<Arg>,
+    object: Option<Arc<dyn Any>>,
 }
 
-#[derive(Debug, Clone)]
-enum Arg {
-    Reusable(Arc<dyn Any>),
-    OneShot(Arc<Mutex<Option<Box<dyn Any>>>>),
-}
+/// `SingleUse` is intended for cases where a command payload should only be
+/// used once; an example would be if you have some resource that cannot be
+/// cloned, and you wish to send it to another widget.
+pub struct SingleUse<T>(Mutex<Option<T>>);
```

Comment:
this type isn's actually exposed anywhere, so I don't think external crates can access it?

Also I would expect this to need to be in an `Arc`, since the `Event` itself still needs to be cloneable?

Reply:
Good catch, I've reexported it from `lib` now.

Not sure what you mean regarding the `Arc`. When using `SingleUse` the payload will be stored the exact same way as it used to with `Arg::OneShot`, minus the `Box<Any>` wrapper as it is now directly stored as `Arc<Any>`.

Reply:
ah okay I missed that the `Arc` has moved into the signature of the `object` field on `Command`.